### DISCUSSION TITLE
Subscription tracking: Not really an issue. More of a question

### DISCUSSION THREAD

**Discussion by petebof** [anycable/anycable#120] [Q&A]
I am migrating a Rails project to AnyCable. One of the issues I am facing is subscription management: I want to know which users unsubscribed from a "global" channel and did not subscribe back within a few minutes. I am thinking of using Redis and a job scheduler for this. 

Is there a better way to do this with ActionCable? Can I get all the users subscribed to a channel?

I found an excellent suggestion by @palkan in a similar "problem": [link](https://github.com/anycable/anycable-rails/issues/12#issuecomment-319362658 )
In the last sentence of this very nice idea,  @palkan mentioned that we need to take care of race conditions in Redis. Since all Redis ops are atomic (and I cannot see any series of Redis ops that need wrapping into a transaction) is locking of Redis ops necessary?

**Comment by palkan** [comment]
> Can I get all the users subscribed to a channel?

There is no built-in feature for that. Presence tracking is beyond AnyCable's scope.

The solution from the link is a very rough sketch and it has an important caveat: `#unsubscribed` / `#disconnect` callbacks are not reliable; we can not guarantee they would be called (for example, in case of a server crash).

A better solution would be to periodically ping server and use some data structure with expiration.

**Reply by petebof** [reply]
Thank you for your reply @palkan ! 
I guess you recommend something based on what you 've written at this [link](https://github.com/anycable/anycable/issues/99#issuecomment-611998267), right?

**Reply by palkan** [reply]
Yep.
